.ci/Jenkinsfile

```
@@ -244,33 +253,93 @@ pipeline {
+  def authObj = props?.data
+  def awsProps = getVaultSecret(secret: "${AWS_PROVISIONER_SECRET}")
+  def awsAuthObj = awsProps?.data
+  withEnv([
+    "ANSIBLE_HOST_KEY_CHECKING=False",
+  ]){
+    withVaultToken(){
+      withEnvMask(vars: [
+        [var: "AWS_ACCESS_KEY_ID", password: awsAuthObj.access_key],
+        [var: "AWS_SECRET_ACCESS_KEY", password: awsAuthObj.secret_key]
+      ]) {
+        sh(script: """#!/bin/bash
+            set -eux
+            if ! test -d ${env.WORKSPACE}/.venv; then
+                python3 -m venv ${env.WORKSPACE}/.venv
+                 ${env.WORKSPACE}/.venv/bin/pip3 -q install wheel
+                 ${env.WORKSPACE}/.venv/bin/pip3 -q install ansible requests google-auth boto3 boto
+            fi
+            ${env.WORKSPACE}/.venv/bin/${args}
+          """,
+           label: "Executing pyrun"
+        )
       }
-      def pullRequestFilter = scenario.containsKey('pullRequestFilter') ? scenario.pullRequestFilter : ''
-      def tags = scenario.tags
-      def regexps = [ "^e2e/_suites/${suite}/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
-      if ("${FORCE_SKIP_GIT_CHECKS}" == "true" || isGitRegionMatch(patterns: regexps, shouldMatchAll: false)) {
-        platformsValue.each { platform ->
-          def platformLabels = platform.labels
+    }
+  }
+}
+
+def ansible(args){
+     pyrun("ansible-playbook -v --private-key=\"${env.WORKSPACE}/${env.BASE_DIR}/e2essh\" --extra-vars=\"workspace=${env.WORKSPACE}/${env.BASE_DIR} runId=${env.RUN_ID.split('-')[0]} sshPublicKey=${env.WORKSPACE}/${env.BASE_DIR}/e2essh.pub\" --ssh-common-args='-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' ${env.WORKSPACE}/${env.BASE_DIR}/.ci/ansible/playbook.yml ${args}")
+}
+
+
+def checkTestSuite(Map parallelTasks = [:], Map item = [:]) {
+  def suite = item.suite
+  def platforms = item.platforms
+
+  if(suite == "fleet"){
+    // Deploy the test infrastructure to fleet related tests
+    sh "ssh-keygen -b 4096 -t rsa -f ${env.WORKSPACE}/${env.BASE_DIR}/e2essh -q -N \"\" "
+    ansible("-t deploy")
+  }
+
+  item.scenarios.eachWithIndex { scenario, scenarioIdx ->
+    def name = scenario.name
+    def platformsValue = platforms
+    def scenarioPlatforms = scenario.platforms
+    if (scenarioPlatforms?.size() > 0) {
+      // scenario platforms take precedence over suite platforms, overriding them
+      platformsValue = scenarioPlatforms
+    }
+    def pullRequestFilter = scenario.containsKey('pullRequestFilter') ? scenario.pullRequestFilter : ''
+    def tags = scenario.tags
+    def regexps = [ "^e2e/_suites/${suite}/.*", "^.ci/.*", "^cli/.*", "^e2e/.*\\.go", "^internal/.*\\.go" ]
+    if ("${FORCE_SKIP_GIT_CHECKS}" == "true" || isGitRegionMatch(patterns: regexps, shouldMatchAll: false)) {
+      platformsValue.each { platform ->
+        def platformLabels = platform.labels
+        if(suite == "fleet" && platformLabels == "arm64" || platformLabels == "amd64"){
```

I think this is more correct 🤔 
```suggestion
        if(suite == "fleet" && (platformLabels == "arm64" || platformLabels == "amd64")){
```